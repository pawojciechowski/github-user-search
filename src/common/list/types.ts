export interface ListItem {
  id: string;
  text: string;
  href?: string;
}
