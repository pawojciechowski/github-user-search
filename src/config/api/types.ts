// TODO: Verify types

export interface GithubUserResponse {
  login: string;
  name: string | null;
  bio: string | null;
  avatar_url: string;
}

export interface GithubReposResponseRepository {
  id: number;
  name: string;
  html_url: string;
  stargazers_count: number;
}

export interface GithubReposResponse extends Array<GithubReposResponseRepository> {}
