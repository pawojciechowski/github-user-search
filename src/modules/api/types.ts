export type APIUrl = string;

export interface IAPIService {
  get<T>(url: APIUrl, params?: { [key: string]: string }): Promise<T>;
}

export interface APIServiceOptions {
  headers?: { [key: string]: string };
  mockResponse?: { [key: string]: any };
  logRequests?: boolean;
}
