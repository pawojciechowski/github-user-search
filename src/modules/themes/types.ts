import { themeNames } from './themes';

export type Color = string;
export type FontSize = string;
export type FontWeight = string;
export type FontFamily = string;
export type LineHeight = string;
export type LetterSpacing = string;

interface FontSettings {
  fontFamily?: FontFamily,
  fontSize?: FontSize,
  fontWeight?: FontWeight,
  lineHeight?: LineHeight,
  letterSpacing?: LetterSpacing,
  color?: Color
}

interface InputStyles {
  placeholder?: {
    text: FontSettings
  },
  background: Color,
  text: FontSettings
}

interface ButtonStyles {
  background: Color,
  text: FontSettings
}

interface TypographySettings {
  base: FontSettings,
  title: FontSettings
}

export interface Theme {
  background: Color,
  input: InputStyles,
  button: ButtonStyles,
  spacer: number,
  typography: TypographySettings,
}

export type ThemeName = typeof themeNames[number];
