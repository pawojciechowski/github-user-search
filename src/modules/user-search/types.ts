export interface GithubUserResponse {
  login: string,
  name: string,
  bio: string | null,
  avatar_url: string
}

export interface GithubReposResponseRepo {
  id: number,
  name: string,
  html_url: string,
  stargazers_count: number
}

export interface GithubReposResponse extends Array<GithubReposResponseRepo> {}
