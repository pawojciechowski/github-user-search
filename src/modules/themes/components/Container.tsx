import styled from 'styled-components';
import { Theme } from '../types';

const Container = styled.div`
  padding-left: ${({ theme }: { theme: Theme }) => `${theme.spacer * 1.5}px`};
  padding-right: ${({ theme }: { theme: Theme }) => `${theme.spacer * 1.5}px`};

  @media screen and (min-width: 768px) {
    max-width: 700px;
    margin: auto;
  }
`;

export default Container;
