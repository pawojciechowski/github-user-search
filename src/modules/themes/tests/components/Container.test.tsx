import React from 'react';
import { render } from '@testing-library/react';
import Container from '../../components/Container';
import { Theme } from '../../types';

const testTheme = {
  spacer: 1,
} as Theme;

describe('Container', () => {
  it('use proper padding based on spacer property in theme', () => {
    const { container } = render(<Container theme={testTheme} />);
    expect(container.firstChild).toHaveStyleRule('padding-left', '1.5px');
    expect(container.firstChild).toHaveStyleRule('padding-right', '1.5px');
  });

  it('matches snapshot', () => {
    const { container } = render(<Container theme={testTheme} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
