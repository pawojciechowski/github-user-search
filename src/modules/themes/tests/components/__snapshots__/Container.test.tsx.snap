// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> matches snapshot 1`] = `
.c0 {
  padding-left: 1.5px;
  padding-right: 1.5px;
}

@media screen and (min-width:768px) {
  .c0 {
    max-width: 700px;
    margin: auto;
  }
}

<div
  class="c0"
/>
`;
